front/src/modules/trainschedule/components/ManageTrainSchedule/hooks/useUpdateTrainSchedule.ts

```diff
@@ -52,40 +68,114 @@ const useUpdateTrainSchedule = (
       rollingStock?.name
     );
 
-    if (formattedSimulationConf && trainIdToEdit) {
-      setIsWorking(true);
-      const trainSchedule = formatTrainSchedulePayload(
-        formattedSimulationConf,
-        confName,
-        startTime
-      );
-      // TODO Paced train : Adapt this to handle paced trains in edit issue of https://github.com/OpenRailAssociation/osrd/issues/10615
-      try {
-        const trainScheduleResult = await putTrainScheduleById({
-          id: formatTrainScheduleIdToEditoastTrainId(trainIdToEdit as TrainScheduleId),
+    if (!formattedSimulationConf || !trainIdToEdit) return;
+
+    setIsWorking(true);
+
+    let editedItemId: TimetableItemId;
+
+    // handle item update without changing type
+    if (
+      !showPacedTrains ||
+      formattedSimulationConf.editingTrainIsPacedTrain === isPacedTrain(trainIdToEdit)
+    ) {
+      let promise;
+      if (isTrainSchedule(trainIdToEdit)) {
+        const trainSchedule = formatTrainSchedulePayload(
+          formattedSimulationConf,
+          confName,
+          startTime
+        );
+        promise = putTrainScheduleById({
+          id: formatTrainScheduleIdToEditoastTrainId(trainIdToEdit),
           trainScheduleForm: trainSchedule,
         }).unwrap();
-        const formattedTrainScheduleResult: TrainScheduleResultWithTrainId = {
-          ...trainScheduleResult,
-          id: formatEditoastTrainIdToTrainScheduleId(trainScheduleResult.id),
-        };
-        upsertTrainSchedules([formattedTrainScheduleResult]);
-        dtoImport();
-        dispatch(
-          setSuccess({
-            title: t('trainUpdated'),
-            text: `${confName}: ${startTime.toLocaleString()}`,
-          })
+      } else {
+        const pacedTrain = formatPacedTrainPayload(formattedSimulationConf);
+        promise = putPacedTrainById({
+          id: formatPacedTrainIdToEditoastTrainId(trainIdToEdit),
+          pacedTrainForm: pacedTrain,
+        }).unwrap();
+      }
+
+      const newTimetableItem = await promise;
+      upsertTimetableItems([
+        {
+          ...newTimetableItem,
+          id: trainIdToEdit,
+        } as TimetableItemWithTimetableId,
```

Comment:
Why do we need the cast here?

Reply:
Casting is required here because typescript doesn't know if the id type (paced/trainSchedule) is the same as that of the object created. I can have a look to try to avoid it

Reply:
Hm I had a look, and I'm not sure we can remove it. We can discuss it if you want